src/client/pfs.go

```diff
@@ -40,11 +40,15 @@ func NewCommitProvenance(repoName string, branchName string, commitID string) *p
 }
 
 // NewFile creates a pfs.File.
-func NewFile(repoName string, branchName string, commitID string, path string) *pfs.File {
-	return &pfs.File{
+func NewFile(repoName string, branchName string, commitID string, path string, tag ...string) *pfs.File {
```

Comment:
This should just be a string.  The empty string has the same effect as this not being set.
Maybe include an example comment showing that it can be `""`.

Reply:
Even if we use this pattern elsewhere because it's convenient, I don't think we should expose it in the public API.  A file does not optionally take a list of tags.

Reply:
Yeah, I kind of have the same intuition about this not being the same as optional configuration. I guess it really is a decision about whether we should support defaults here, or require explicit specification of each of the fields. We could have the default file and tag be empty string since those are supported defaults by several operations. I do like keeping the tag optional though because most users should not need to think about tags when operating on files.

Reply:
If someone sees a parameter they don't recognize, they will probably look at the function documentation and see that it is okay to pass `""`.  A functional option would be fine here too.  But the variadic tags argument is presenting the wrong model to users.

Reply:
Okay, I can see that. My general approach to adding new optional parameters to functions that are heavily used is to start with a variadic parameter for the new parameter then functional options if more are needed. We should probably get some more opinions about this considering how heavily used this will be. @jdoliner @Tryneus what do you guys think the signature of this function should look like? Should these just all be required or would having files and / or tags be functional options make sense? It might be a bit weird to have the file path as a functional option, but using the tag should be very rare for most users.

Reply:
We're currently doing some rework on the way these structures are built in https://github.com/pachyderm/pachyderm/pull/6152 so this may need some of the same treatment.  In general, I'm negative on using a variadic argument for non-variadic data (but I also don't like golang zero-values).  I think after #6152, since people will usually be building files off of commits, the common use case will be `commit.NewFile("path", "tag") which is less of a magical invocation of a tsunami of strings.  I would lean slightly towards not using a variadic argument, though.

Reply:
Okay, I am just going to remove the tag field from this constructor for now and readd it as a required parameter once those changes are in.

Reply:
Coming in late to this thread but I like the decision here.